resources/packs/gcp/gcp.lr

```diff
@@ -1143,3 +1145,104 @@ private gcp.project.pubsubService.snapshot @defaults("name") {
   // When the snapshot expires
   expiration time
 }
+// GCP KMS resources
+private gcp.project.kmsService {
+  // Project ID
+  projectId string
+  // Available locations for the service
+  locations() []string
+  // List of keyrings in the current project
+  keyrings() []gcp.project.kmsService.keyring
+}
+
+// GCP KMS keyring
+private gcp.project.kmsService.keyring @defaults("name") {
+  // Project ID
+  projectId string
+  // Keyring name
+  name string
+  // Time created
+  created time
+  // Keyring location
+  location string
+  // List of cryptokeys in the current keyring
+  cryptokeys() []gcp.project.kmsService.keyring.cryptokey
+}
+
+// GCP KMS crypto key
+private gcp.project.kmsService.keyring.cryptokey @defaults("name purpose"){
+  // Crypto key name
+  name string
+  // Primary version for encrypt to use for this crypto key
+  primary gcp.project.kmsService.keyring.cryptokey.version
+  // Crypto key purpose
+  purpose string
+  // List of cryptokey versions
+  versions() []gcp.project.kmsService.keyring.cryptokey.version
+}
+
+// GCP KMS crypto key version
+private gcp.project.kmsService.keyring.cryptokey.version @defaults("name state") {
+  // Crypto key version name
+  name string
+  // Crypto key version's current state
+  state string
+  // The protection level describing how crypto operations perform with this crypto key version
+  protectionLevel string
+  // Algorithm that this crypto key version supports
+  algorithm string
+  // Statement generated and signed by HSM at key creation time
+  attestation gcp.project.kmsService.keyring.cryptokey.version.attestation
+  // Time created
+  created time
+  // Time generated
+  generated time
+  // Time destroyed
+  destroyed time
+  // Destroy event timestamp
+  destroyEventTime time
+  // Name of the import job used in the most recent import of this crypto key version
+  importJob string
+  // Time at which this crypto key version's key material was imported
+  importTime time
+  // The root cause of an import failure
+  importFailureReason string
+  // Additional fields for configuring external protection level
+  externalProtectionLevelOptions gcp.project.kmsService.keyring.cryptokey.version.externalProtectionLevelOptions
+  // Whether the crypto key version is eligible for reimport
+  reimportEligible bool
+}
+
+// GCP KMS crypto key version attestation
+private gcp.project.kmsService.keyring.cryptokey.version.attestation {
+  // Crypto key version name
+  cryptoKeyVersionName string
+  // Format of the attestation data
+  format string
+  // Attestation data provided by the HSM when the key version was created
+  // content ???
```

Comment:
@chris-rock do we want to do anything with this? In my opinion, this isn't relevant since it's just bytes and you cannot really do anything with it in MQL

Reply:
Lets not expose this for now